startables/readers/word.py

```diff
@@ -0,0 +1,72 @@
+import os
+import re
+from pathlib import Path
+from typing import Union
+
+import pandas as pd
+
+from startables import Bundle, ColumnMetadata, Table
+from startables.startables import TableOrigin
+
+
+TABLE_BLOCK_MARKER_PATTERN = r'^\*\*[^\*].*'
+
+
+def import_from_word(path: Union[str, Path]) -> Bundle:
+    """
+    Imports table blocks from tables in an MS Word docx file.
+    Only reads those Word tables whose top left cell is a table block start marker ('**table_name').
+
+    MS Word documents cannot strictly live up to the StarTable format specification and therefore
+    this function is not to be considered a formal reader. Rather, it is a utility that
+    merely attempts to import table blocks.
+
+    :param path: The path to the docx file
+    :return: Bundle containing the imported tables
+    """
+    try:
+        import docx
+    except ImportError:
+        raise ImportError("Missing optional dependency 'docx'. Install python-docx package "
+                          "for MS Word support. Use pip or conda to install python-docx.") from None
+
+    if not os.path.exists(str(path)):
```

Comment:
Would it be cleaner to always convert `path` to `Path` to only get one code-path for the path gymnastics? Especially since `pathlib` is much more ergonomic that `os.path`? We could make our own `aspath` in parallel to `np.asarray`?